framework/src/variables/MooseVariableFV.C

```diff
@@ -491,50 +508,40 @@ MooseVariableFV<OutputType>::getNeighborValue(const Elem * const neighbor,
   if (neighbor && this->hasBlocks(neighbor->subdomain_id()))
     return getElemValue(neighbor);
   else
-  {
-    // If we don't have a neighbor, then we're along a boundary, and we may have a DirichletBC
-    const auto & pr = getDirichletBC(fi);
-
-    if (pr.first)
-    {
-      mooseAssert(pr.second, "The FVDirichletBC is null!");
-
-      const FVDirichletBCBase & bc = *pr.second;
+    // If we don't have a neighbor, then we're along a boundary
+    // Linear interpolation: face_value = (elem_value + neighbor_value) / 2. Note that weights of
+    // 1/2 are perfectly appropriate here because we can arbitrarily put our ghost cell centroid
+    // anywhere and by convention we locate it such that a line drawn between the ghost cell
+    // centroid and the element centroid is perfectly bisected by the face centroid
+    return 2. * getBoundaryFaceValue(fi) - elem_value;
+}
 
-      // Linear interpolation: face_value = (elem_value + neighbor_value) / 2
-      return 2. * bc.boundaryValue(fi) - elem_value;
-    }
-    else
-      // No DirichletBC so we'll implicitly apply a zero gradient condition and assume that the
-      // face value is equivalent to the element value
-      return elem_value;
-  }
+template <typename OutputType>
+bool
+MooseVariableFV<OutputType>::isInternalFace(const FaceInfo & fi) const
+{
+  const bool is_internal_face = fi.faceType(this->name()) == FaceInfo::VarFaceNeighbors::BOTH;
+  mooseAssert(is_internal_face == (this->hasBlocks(fi.elem().subdomain_id()) && fi.neighborPtr() &&
+                                   this->hasBlocks(fi.neighborPtr()->subdomain_id())),
+              "Sanity checking whether we are indeed an internal face");
+  return is_internal_face;
 }
 
 template <typename OutputType>
-ADReal
-MooseVariableFV<OutputType>::getFaceValue(const Elem * const neighbor,
-                                          const FaceInfo & fi,
-                                          const ADReal & elem_value) const
+const ADReal &
+MooseVariableFV<OutputType>::getInternalFaceValue(const Elem * const neighbor,
+                                                  const FaceInfo & fi,
+                                                  const ADReal & elem_value) const
 {
-  // Are we on a boundary or an interface beyond which our variable doesn't exist?
-  if (!neighbor || !this->hasBlocks(neighbor->subdomain_id()))
-  {
-    const auto & pr = getDirichletBC(fi);
+  mooseAssert(isInternalFace(fi), "This function only be called on internal faces.");
 
-    if (pr.first)
-    {
-      const FVDirichletBCBase & bc = *pr.second;
+  auto pr = _face_to_value.emplace(&fi, 0);
```

Comment:
yes

Reply:
ok good to know. @pbalest was advocating for that and I did not know we were doing this. 
Maybe we should store the outcome of RC interpolation too then?

Reply:
this is option 1 in #16585

Reply:
Storing the outcome of RC interpolation is kind of a different beast conceptually I think. It's outside the traditional variable data, e.g. it is the result of information from multiple variables. My mind goes naturally to something like an `INSFVRhieChowUserObject` (or perhans even `RhieChowUserObject` with optionally `typedef RhieChowUserObject INSFVRhieChowUserObject` since it sounds like we *will* be using Rhie-Chow for compressible physics). This user object would be the object that caches Rhie-Chow data, which could be used in kernels and postprocessors

Reply:
Maybe we also create a base `InterpolationUserObject` that has a functor attribute that can be set like @rwcarlsen likes or a virtual interface like @rwcarlsen hates.

The idea behind the `enum class InterpMethod` and the global function `Moose::FV::interpolate` was a noble one, but the issue is that it's not extensible, e.g. a module or application developer cannot modify the list of enumerators.

Reply:
Agreed that it's information from multiple variables, however as far as the simulation is concerned, it's the velocity value on the face. At least for now, the only kernels that compute the velocity on the face are the advection kernels and they use RC. I dont know if any other kernel would deliberately use something else but I suspect it'd bring numerical oddities. 

The user object works. Adding INSFVUtils to house a ::RCInterpolate would also work. I added it to #16585

Reply:
ok for not putting in a quick fix by storing velocity face values